server/auth.go

```diff
@@ -57,3 +59,39 @@ func authenticateStream(authURL *url.URL, incomingRequestURL string) (*authWebho
 
 	return &authResp, nil
 }
+
+func getTranscodeConfiguration(r *http.Request) (*authWebhookResponse, error) {
+	transcodeConfigurationHeader := r.Header.Get(LIVERPEER_TRANSCODE_CONFIG_HEADER)
+	if transcodeConfigurationHeader == "" {
+		return nil, nil
+	}
+
+	var transcodeConfiguration authWebhookResponse
+	err := json.Unmarshal([]byte(transcodeConfigurationHeader), &transcodeConfiguration)
+
+	return &transcodeConfiguration, err
+}
+
+// Compare two sets of profiles. Since there's no deep equality method in Go,
+// we marshal to JSON and compare the resulting strings
+func (a authWebhookResponse) areProfilesEqual(b authWebhookResponse) bool {
+	// Return quickly in simple cases without trying to marshal JSON
+	if len(a.Profiles) != len(b.Profiles) {
+		return false
+	}
+	if len(a.Profiles) == 0 {
+		return true
+	}
+
+	profilesA, err := json.Marshal(a.Profiles)
```

Comment:
Nitpick: is it ok to suppress error here?

Reply:
Yeah, I can't see any actual situation where this wouldn't marshal but probably good practice to do something with the error anyway